clouddq/classes/dq_configs_cache.py

```diff
@@ -534,7 +534,7 @@ def is_dataplex_entity(
             data_path=entity_uri.get_entity_id(),
         )
         logger.info(f"Retrieved Dataplex Entities:\n{pformat(dataplex_entities_match)}")
-        if len(dataplex_entities_match) != 1:
+        if len(dataplex_entities_match) == 0:
```

Comment:
Line 548: "dataplex_entity = dataplex_entities_match[0]"

If we expect the user to have the full/exact data_path (e.g. bigquery://) in `entity_uri` when they define the rule, shouldn't we expect 0 or exactly 1 entity to be found? Wondering why we expect >1 matched entities...

Reply:
You are right this should be changed to expect exact matches and have different error handling branches for 0 records found vs multiple records found.